### PR TITLE
Add esp wifi to monorepo with history

### PR DESCRIPTION
**Pull request by bjoernQ**
I also kept all the commits because
- the original authors should be mentioned
- the history might be useful

This is not just a copy from esp-wifi since some changes were needed to get it working with current `main` - however most changes should be straight forward

### CODE REVIEW

**Comment by jessebraham** [comment]
All our PRs get squashed by the merge queue FYI, if you're not aware.

**Comment by bjoernQ** [comment]
> All our PRs get squashed by the merge queue FYI, if you're not aware.

Ah true ... but at least the contributors will be added to that commit

**Comment by MabezDev** [comment]
Hmm, so I guess the fun here will be that we need to make esp-wifi (and its examples) run on stable :D.

Other than that this is looking good so far.

File: esp-hal/src/soc/esp32/mod.rs [hunk 1]
```diff
@@ -18,13 +18,14 @@ pub mod peripherals;
 pub mod psram;
 pub mod radio_clocks;
 
+#[macro_export]
```
[the remark this answers is not in the document]
Comment (bjoernQ) [reply esp-hal/src/soc/esp32/mod.rs:21]: That macro might be useful to users - but then I guess I should add docs

**Comment by bjoernQ** [comment]
> One small request, could you please also add EspWifi to the Package enum in the xtask package? I keep forgetting to do this when I add packages, so trying to build the habit for myself and others 😅

Thanks for the hint .... I totally forgot about that 👍

File: esp-hal/src/soc/esp32/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jessebraham) [reply esp-hal/src/soc/esp32/mod.rs:21]: Sure, either way, needs something done 😁 Thanks

File: Cargo.toml [hunk 2]
```diff
@@ -13,6 +13,9 @@ exclude  = [
     "esp-metadata",
     "esp-println",
     "esp-riscv-rt",
+    "esp-wifi",
     "examples",
+    "extras/bench-server",
+    "extras/esp-wifishark",
```
[the remark this answers is not in the document]
Comment (bjoernQ) [reply Cargo.toml:19]: Definitely yes - I plan to do that in a separate PR 👍

**Comment by maxwen** [comment]
Thx! Reducing the dependency hell using main :)

**Pull request merged**